### PR TITLE
Feat/v4 Twilio iOS v5

### PR DESCRIPTION
**Pull request by lonnylot**
Upgrading to use Twilio 5.x iOS SDK.

This is an important update as the Twilio iOS 2.x SDK is unsupported as of January 1st 2020 and they recommend using the iOS 5.x SDK for incoming calls on iOS >= 13.x.

I am not an iOS dev. and I am giving this my best shot. I believe there are bugs, but I wanted to get it out publicly to ask for help from anyone who knows iOS/Objective-C.

EDIT: Here are some links related to the breaking changes:

1. https://support.twilio.com/hc/en-us/articles/360035005593-iOS-13-Breaking-Changes
1. https://github.com/twilio/voice-quickstart-swift/issues/251

### CODE REVIEW

**Comment by enaluz** [comment]
After some initial testing on my app, it seems that when the call is ended on either end of the call, the call isn't terminated for the other party. Perhaps a bug with the `performEndCallAction` method on line 543? Will try to investigate further.

**Comment by lonnylot** [comment]
Ethan, please make sure you have the latest version updated about 2 hours ago. I believe I found and corrected the issue.

**Comment by lonnylot** [comment]
Put correctly: I found an issue with canceling incoming calls. Make sure you have that fix when testing and let me know if that fixes your issue.

**Comment by enaluz** [comment]
That fix resolved the issue! Everything seems to check out on my end so far. The only small issue I found was that at the top of `performEndCallActionWithUUID`, there used to be a conditional check for the `uuid` that would just `return` if `uuid == nil`. An exception would be raised at the JS layer because that check was removed in your changes, however it wouldn't crash the app. 

Not sure how important the uuid is (and in what situations it would be important), but my app is taking calls successfully without having the uuid set. Other than that, everything seems to check out in dev for me!

**Comment by enaluz** [comment]
Upon closer investigation, some of the event listeners are not working in the JS layer. I'll update this comment with more details in a bit.

**Comment by lonnylot** [comment]
Fixes #89

File: README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ Edit your `Podfile` to include TwilioVoice and RNTwilioVoice frameworks
 source 'https://github.com/cocoapods/specs'
 
 # min version for TwilioVoice to work
-platform :ios, '8.1'
+platform :ios, '9.0'
```
Comment (fabriziomoscon) [inline README.md]: It this correct? Our should it be 10.0?

File: ios/RNTwilioVoice/RNTwilioVoice.m [hunk 2]
```diff
@@ -210,113 +227,149 @@ - (NSString *)fetchAccessToken {
...
-  self.callInvite = callInvite;
+    /**
+     * Calling `[TwilioVoice handleNotification:delegate:]` will synchronously process your notification payload and
+     * provide you a `TVOCallInvite` object. Report the incoming call to CallKit upon receiving this callback.
+     */
+
+    NSLog(@"callInviteReceived:");
 
-  [self reportIncomingCallFrom:callInvite.from withUUID:callInvite.uuid];
+    NSString *from = @"Voice Bot";
```
Comment (fabriziomoscon) [inline ios/RNTwilioVoice/RNTwilioVoice.m]: What is Voice Bot?
Should this be something like `unknown`?

**Comment by enaluz** [comment]
After playing around with this a bit, I found a bug when doing back to back calls. The second call doesn't terminate for the callee when the caller terminates the call. In the second call, two logs are missing that showed up during the first call: `provider:performEndCallAction` and  `provider:didDeactivateAudioSession`. Seems like an easy fix, will try to fix this on my end today or tomorrow.

**Comment by enaluz** [comment]
The bug I described above was introduced in part, because of `self.userInitiatedDisconnect = YES;` in line 121 of `RNTwilioVoice.m`. This change broke for me because in the JS layer I was executing the `disconnect` method under the `connectionDidDisconnect` event listener (I don't use a CallUI in my app, I just use the CallKit UI). Although removing the `disconnect` method in the JS layer would make this PR work for me, I believe this would still break for other people. Not sure what the best move is in this case. 

Thoughts @fabriziomoscon @gitstud ?

Happy to clarify if this was a vague or confusing explanation.

**Comment by fabriziomoscon** [comment]
I will try to verify this and help during the weekend. Unfortunately my app conversion to RN 0.61 is taking forever, but once I am in a position to make and receive calls I will help 100% with implantation

**Comment by lonnylot** [comment]
@enaluz May I ask why you're calling the disconnect method from the `connectionDidDisconnect` event?

Also, could you please provide the NSLogs so we can see what code path is being taken? I suspect it's from `activeCall` or `activeCallInvites` not being cleared at the appropriate time.

**Comment by lonnylot** [comment]
I finally finished my dialer yesterday and was able to test all of the events.

@fabriziomoscon what can I do to help get this committed so we can continue to bug fix and add helpful features to this library?

**Comment by lonnylot** [comment]
Yeah, that's fare. Sorry about that - the initial implementation was just me copying the updated Twilio QuickStart, which is probably the wrong way to do it.

I'll update it again with just the changes required for updating to the new version of the iOS SDK.

**Comment by fabriziomoscon** [comment]
This was automatically closed by me deleting the base branch `feature/android-skd-4`.
The new base branch is: https://github.com/hoxfon/react-native-twilio-programmable-voice/pull/144

Please do not remove https://github.com/lonnylot/react-native-twilio-programmable-voice/tree/feat/v4-ios-v5 as I will include it into #144

**Pull request closed without merging**